File: set_provider.js
const { exec } = require('child_process')

async function main() {
    let nodeEndpoints = ["/ip4/127.0.0.1/tcp/40837/p2p/12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X"];
    let port = 40838;

    //console.log('cargo run -- --listen-address /ip4/127.0.0.1/tcp/40837 --secret-key-seed 1 provide');
    /*
    exec('cargo run -- --listen-address /ip4/127.0.0.1/tcp/40837 --secret-key-seed 1 provide', (err, stdout, stderr) => {
            if (err) {
                console.log(`stderr: ${stderr}`)
                return
            }
            console.log(`stdout: ${stdout}`)
        }
    )*/

    const _sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
    await _sleep(2000);

    for (let i = 0; i < 40; i++) {

        let command = `cargo run -- --peer ${nodeEndpoints[0]} --listen-address /ip4/127.0.0.1/tcp/${port} --group ${i} provide`;

        console.log(command);
        exec(command, (err, stdout, stderr) => {
                if (err) {
                    console.log(`stderr: ${stderr}`)
                    return
                }
                console.log(`stdout: ${stdout}`)
            }
        )

        port += 1;
    }


}

main();

File: src/set_provider.rs
use std::process::{Child, Command, Stdio};
use std::os::unix::io::{AsRawFd, FromRawFd};

fn stdout_to_stdin(process: &Child) -> Option<Stdio> {
    if let Some(ref stdout) = process.stdout {
        return Some(unsafe { Stdio::from_raw_fd(stdout.as_raw_fd()) });
    }
    None
}

fn main() {
    // Bootstrap
    Command::new("cargo")
        .arg("run")
        .arg("--")
        .arg("--listen-address")
        .arg("/ip4/127.0.0.1/tcp/40837")
        .arg("--secret-key-seed")
        .arg("1")
        .arg("provide")
        //.stdout(Stdio::piped())
        .spawn().expect("failed to start");


    let port = 40838;
    for i in 2..3 {
        Command::new("cargo")
            .arg("run")
            .arg("--")
            .arg("--peer")
            .arg("/ip4/127.0.0.1/tcp/40837/p2p/12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X")
            .arg("--listen-address")
            .arg(format!("/ip4/127.0.0.1/tcp/{}", port + i))
            .arg("--secret-key-seed")
            .arg(i.to_string())
            .arg("provide")
            //.stdout(Stdio::piped())
            .spawn().expect("failed to start");
    }

}